### PR TITLE
fix:  Fuzzy & operator usage when handling Intelligent Search API requests

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

This PR aims to correct fuzzy and operator usage when handling Intelligent Search API requests.

Previously, we specified a default value for fuzzy `fuzzy=auto`. But we shouldn’t include it on the first request to Search.  According to the IS team, the correct behavior should let the search engine decide the best parameters for the search.

## How it works?

At the beginning of a search, requests should not define fuzzy and operator parameters. The request response will return the parameters selected for this search.

Example: 

**Request**  
 `https://storeframework.vtexcommercestable.com.br/api/io/_v/api/intelligent-search/facets/trade-policy/1?page=1&count=12&query=awe&sort=&locale=en-US&hideUnavailableItems=false`

**Response** 
recordsFiltered 792,

Then, we should apply these parameters when refining the search result; in our context, it should cover two cases:
* Pagination 
* Filters Usage (facets, sort)

**Request**  
`https://storeframework.vtexcommercestable.com.br/api/io/_v/api/intelligent-search/product_search/trade-policy/1?page=1&count=12&query=awe&sort=&locale=en-US&hideUnavailableItems=false&fuzzy=auto&operator=and`

The implemented approach here involves passing the necessary parameters, in the form of facets, after the initial request and using their structure.

## How to test it?

Use this [preview link ](https://sfj-320f502--starter.preview.vtex.app/) to test the following scenarios:

** scenario 1: Search Catergory**
1. Click on `office` category
2. Open the inspector -> Network tab, filter for `ClientProductGalleryQuery`
3. In the `preview` tab, search for the `metadata` field in data
4. The `fuzzy` and `logicalOperator` values should correspond to the values in the url, ex: `?category-1=office&fuzzy=0&operator=and&`

image

5. Try adding another filter and add a new page (`Load More Products` button), the parameters should persist.

** scenario  2: Search for a term**
1. In the SearchInput type `awe` and enter
2. Repeat the steps from the previous scenario (start at step 2).

### Starters Deploy Preview

https://sfj-320f502--starter.preview.vtex.app/

## References

- [VTEX Intelligent Search - Fuzzy ](https://help.vtex.com/tracks/vtex-intelligent-search--19wrbB7nEQcmwzDPl1l4Cb/23mytRDsEduqLO0Lo7yufy#configuring-autocorrect)
- [Intelligent Search API - PLP](https://developers.vtex.com/docs/api-reference/intelligent-search-api#get-/product_search/-facets-)

### CODE REVIEW

File: packages/core/src/sdk/product/useProductGalleryQuery.ts
Comment (eduardoformiga) [inline packages/core/src/sdk/product/useProductGalleryQuery.ts:86]: could the `term` influence this logic?  maybe we could move the `if(term){...}` just for the rest of the logic internally (sendAnalyticsEvent).

**Review by eduardoformiga — COMMENTED** [review]
Looks Good! I'll test it in practice before approval, but I left some comments so far.

File: packages/api/src/platforms/vtex/clients/search/index.ts
Comment (hellofanny) [reply packages/api/src/platforms/vtex/clients/search/index.ts:33]: my bad, I should've removed it while testing.

File: packages/core/src/sdk/product/useProductGalleryQuery.ts
Comment (eduardoformiga) [inline packages/core/src/sdk/product/useProductGalleryQuery.ts:1]: should we also add this logic in the `useSuggestions` hook that also uses `onSuccess` with `metadata`?

File: packages/core/src/sdk/product/useProductGalleryQuery.ts
Comment (hellofanny) [reply packages/core/src/sdk/product/useProductGalleryQuery.ts:86]: it does, because we just want to send this parameters when we are using the search (looking for a term). We shouldn't send it when going directly the category PLP.

**Review by lucasfp13 — COMMENTED** [review]
LGTM 🚀
But I'll wait for [the reply here](https://github.com/vtex/faststore/pull/2204/files#r1466827719), it makes sense to me but idk if in fact it does 😬

File: packages/core/src/sdk/product/useProductGalleryQuery.ts
Comment (hellofanny) [reply packages/core/src/sdk/product/useProductGalleryQuery.ts:1]: well, I’m not sure, but as far as I tried and tested, applying on useProductGalleryQuery cover the needed cases.
useTopSearch (we don’t need the parameters info here, because we're only listing) will lead us to useProductGalleryQuery, the same goes for useSuggestions. 
We only need to apply the parameters when a refinement is needed, for our scenario only applies inside the product gallery context. 🤔

Comment (eduardoformiga) [reply packages/core/src/sdk/product/useProductGalleryQuery.ts:1]: It makes sense to me!

File: packages/core/src/sdk/product/useProductGalleryQuery.ts
Comment (eduardoformiga) [inline packages/core/src/sdk/product/useProductGalleryQuery.ts:90]: I noticed that the page is rendering twice. 

PLP: 
1. First load: /office
2. second load: /office?category-1=office&fuzzy=0&operator=and&facets=category-1%2Cfuzzy%2Coperator&sort=score_desc&page=0

Is there a way to set it without rendering twice?

**Review by eduardoformiga — COMMENTED** [review]
.

File: packages/api/src/platforms/vtex/clients/search/index.ts
Comment (lucasfp13) [inline packages/api/src/platforms/vtex/clients/search/index.ts:77]: Nice!!

**Review by lucasfp13 — APPROVED** [review]
LGTM!

File: packages/core/src/sdk/product/useProductGalleryQuery.ts
Comment (hellofanny) [reply packages/core/src/sdk/product/useProductGalleryQuery.ts:93]: I've tried to solve this issue, but couldn't find a workaround for it :/ 

The team agreed to move forward with the upon approach and revisit it later.

**Pull request merged**
